alonzo/formal-spec/txinfo.tex

```diff
@@ -3,5 +3,232 @@ \section{TxInfo Construction}
+    inputs (ie. its realized inputs). This is done in order to maintain
+    deterministic evaluation. For details, see~\ref{sec:txinfo}.
+
+    $\fun{valContext}$ constructs the \emph{validation context}, also referred to as the
+    script context. A validation context is
+    a $\Data$ term which encodes both the summary of the transaction and ledger information
+    (this is supplied by the $\fun{txInfo}$ summarization function), and the script purpose.
+
+\begin{figure}
+  \emph{Plutus Library Types and Functions}
+  %
+  \begin{align*}
+      &\type{P.ScriptContext} ~=~ \type{P.TxInfo} \times \type{P.ScriptPurpose} \\
+      &\text{Plutus Script Context}
+      \nextdef
+      &\fun{P.toData} \in \type{P.T} \to \Data \\
+      &\text{Constructs a Data term from a Plutus term of type P.T}
+  \end{align*}
+    \emph{Ledger Functions}
+    %
+    \begin{align*}
+        &\fun{txInfo} \in \Language \to \PParams \to \UTxO \to \Tx \to \TxInfo \\
+        &\text{Summarizes transaction data}
+        \nextdef
+        &\fun{valContext} \in \type{P.TxInfo} \to \ScriptPurpose \to \Data \\
+        &\fun{valContext}~txinfo~sp = \fun{P.toData}~(txinfo,~sp_{P})
+        &\text{Pairs transaction data with a script purpose}
+    \end{align*}
+  \caption{Transaction Summarization Functions}
+  \label{fig:txinfo-funcs}
+\end{figure}
+
+We give the Plutus types corresponding to the ledger counterparts in Figures \ref{fig:txinfo-types}
+and \ref{fig:txinfo-types-two}. Details for types that have non-identity translation
+functions between ledger and Plutus types are in Figure \ref{fig:txinfo-translations}.
+
+Certain types, such as the bootstrap address type, cannot be passed to scripts, and
+are therefore translated as $\Nothing$. All terms made up of untranslatable
+terms also translate to $\Nothing$. For example, since a bootstrap address $a \in \AddrBS$
+is not translatable, neither is $a\in\Addr$, and $(a,\wcard,\wcard)~\in~\TxOut$
+also translates to $\Nothing$.
+
+We denote a term $t$ of a ledger type $\type{LT}$ that has been translated to the
+corresponding Plutus type by $t_P$.
+
+Translating certain kinds of certificates drops the data in the
+certificates, in particular, the $\DCertGen$ and $\DCertMir$ ones.
+
+The constant returned by $\fun{slotToPOSIXTime}$,
+
+\[\type{ResolutionConstant}\]
+
+Is the maximum possible value of a POSIX time interval. This will be adjusted in
+future eras to reflect the time being translated.
+
+Note that the $\Ptr$ addresses translated and passed to Plutus scripts are
+not resolvable to their corresponding key or script staking credencials. This
+is because doing a lookup in the resolution map, which part of the ledger state,
+could break determinism.
+
+During the transaction encoding and decoding process, a transaction is discarded if it is not
+encoded correctly. This includes, in particular, a check that scripts and $\Data$
```

Comment:
Either $\Script$ and $\Data$ elements, or scripts and datums, but don't mix them